src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -193,6 +230,35 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         });
     }
 
+    private  onToolbarItemSelected(toolbarItem: ToolbarItem) {
+        switch (toolbarItem.props.name) {
+            case "drawRectangle":
```

Comment:
Instead of calling the same method over and over you can set a var with the SelectionMode then call the method once at the end of the switch block.